apps/wallet-mobile/src/components/hooks/useStatusBar.ts

```diff
@@ -0,0 +1,72 @@
+import {useFocusEffect} from '@react-navigation/native'
+import {useTheme} from '@yoroi/theme'
+import * as React from 'react'
+import {Platform, StatusBar as StatusBarRN} from 'react-native'
+
+import {HexColor} from '../../theme/types'
+import {useModal} from '../Modal/ModalContext'
+
+export const useStatusBar = (baseColor?: HexColor, legacyModal?: boolean) => {
+  const {theme, isDark} = useTheme()
+  const {isOpen} = useModal()
+
+  const reflectStatusBarColor = React.useCallback(() => {
+    const color = baseColor ?? (isDark ? theme.color['black-static'] : theme.color['white-static'])
+    const bgColor = isOpen || legacyModal ? simulateOpacity(color) : color
+
+    StatusBarRN.setBarStyle(isDark ? 'light-content' : 'dark-content', true)
+    if (Platform.OS === 'android') StatusBarRN.setBackgroundColor(bgColor, true)
+  }, [baseColor, isDark, isOpen, legacyModal, theme.color])
+
+  useFocusEffect(reflectStatusBarColor)
+}
+
+/**
+ *
+ * @description Returns a color with half intensity, to simulate opacity, used to mimic the modal backdrop on top of it, I.e rgba(0,0,0,0.5), it will ignore the alpha channel if it was passed along with the color
+ * @param color Hex color in "#224466" | "#22446620" | #246" | "#2468" format
+ * @returns Hex color halved "#112233" | "#11223320" if the input is correct, otherwise it will return the same color passed
+ * @warning This function will ignore the alpha channel if it was passed along with the color
+ * @warning This function ignore named colors, by ignoring, it means returning the same color passed
+ */
+export const simulateOpacity = (color: HexColor): HexColor => {
+  if (!/^#([0-9A-Fa-f]{3}([0-9A-Fa-f]{1})?|[0-9A-Fa-f]{6}([0-9A-Fa-f]{2})?)$/.test(color)) {
+    return color
```

Comment:
not throwing cuz it can't figure out if it is a named color or an invalid color, therefore returning as it is, the effect would be to don't opaque the StatusBar, which is not the end of the world